plugins/hls-code-range-plugin/test/Main.hs

```diff
@@ -48,6 +48,7 @@ selectionRangeGoldenTest testName positions = goldenGitDiff testName (testDataDi
         let res = resp ^. result
         pure $ fmap (showSelectionRangesForTest . absorbNull) res
     case res of
+        Left (ResponseError (InL LSPErrorCodes_ContentModified) _ _) -> pure ""
```

Comment:
seems suspicious????

Reply:
The test was assuming that if there is no response they should get back an empty result, but we now handle empty responses with errors instead.

Reply:
Hmmm, is that right? You don't mean empty result as in "the plugin correctly computed the results and it happened to be empty" right? because that probably should be returned as an empty result.

Reply:
Nope, a lot of plugins used to (and some still do!) handle failures by just returning an empty result. The reason they did this was to avoid having popups when the data they wanted wasn't ready. For here the specific PR is https://github.com/haskell/haskell-language-server/pull/3229 . This is also the reason we return rule and dependency failures with ContentModified because we don't want to show popups unless there's actually an error.

Reply:
I still don't understand, why do we produce a `LSPErrorCodes_ContentModified` for the empty result in this plugin? Is it because the only reason why we might produce an empty result is because we have modified the content before resolving `SMethod_TextDocumentSelectionRange `?

Reply:
Previously it was because that's how we treated all rule failures.. now we will respond with a RequestFailed response, so we need to change that here to consider RequestFailed as nothing.